### PR TITLE
Add new signals to Multigrid

### PR DESCRIPTION
**Pull request by peterrum**
I am not sure about the names. 

~depends on #12530~

### CODE REVIEW

File: include/deal.II/multigrid/multigrid.h [hunk 1]
```diff
@@ -118,11 +118,25 @@ namespace mg
...
     boost::signals2::signal<void(const bool before, const unsigned int level)>
       post_smoother_step;
+
+    /**
+     * This signal is triggered before (@p before is true) and after (@p before
+     * is false) the computation of the risiduum vector on @p level, including
+     * the result of edge_out and edge_down.
+     */
+    boost::signals2::signal<void(const bool before, const unsigned int level)>
+      residuum_step;
```
Comment (tjhei) [inline include/deal.II/multigrid/multigrid.h:132]:
```suggestion
      residual_step;
```
?

Comment (tjhei) [reply include/deal.II/multigrid/multigrid.h:132]: or maybe just ``residual``?

File: include/deal.II/multigrid/multigrid.h [hunk 2]
```diff
@@ -118,11 +118,25 @@ namespace mg
...
+     * This signal is triggered before (@p before is true) and after (@p before
+     * is false) the computation of the residual vector on @p level, including
+     * the result of edge_out and edge_down.
+     */
+    boost::signals2::signal<void(const bool before, const unsigned int level)>
+      residual_step;
+
+    /**
+     * This signal is triggered before (@p before is true) and after (@p before
+     * is false) the execution of edge_in and edge_up..
```
Comment (tjhei) [inline include/deal.II/multigrid/multigrid.h:136]:
```suggestion
     * is false) the execution of edge_in and edge_up.
```

Comment (peterrum) [reply include/deal.II/multigrid/multigrid.h]: Done :)

**Review by kronbichler — APPROVED** [review]

**Pull request merged**